test/DatasetAuthorization.js

```diff
@@ -633,4 +645,922 @@ describe("0300: DatasetAuthorization: Test access to dataset", () => {
+    const invalidDatasetWithInvalidPid = {
+      ...TestData.RawWrong_1,
+      pid: "this-is-invalid-pid-1",
+      ownerGroup: "group1",
+    };
+
+    return request(appUrl)
+      .post("/api/v3/Datasets")
+      .send(invalidDatasetWithInvalidPid)
+      .set("Accept", "application/json")
+      .set({ Authorization: `Bearer ${accessTokenAdmin}` })
+      .expect("Content-Type", /json/)
+      .then((res) => {
+        res.statusCode.should.not.be.equal(200);
+        res.body.should.not.have.property("pid");
+      });
+  });
+
+  it("0600: user with create dataset groups only can add a new raw dataset", async () => {
+    const newDataset = {
+      ...TestData.RawCorrect,
+      ownerGroup: "group1",
+    };
+
+    return request(appUrl)
+      .post("/api/v3/Datasets")
+      .send(newDataset)
+      .set("Accept", "application/json")
+      .set({ Authorization: `Bearer ${accessTokenUser1}` })
+      .expect(200)
+      .expect("Content-Type", /json/)
+      .then((res) => {
+        res.body.should.have.property("pid").and.be.string;
+        res.body.should.have.property("owner").and.be.string;
+        res.body.should.have.property("type").and.equal("raw");
+      });
+  });
+
+  it("0601: user with create dataset groups only cannot add a new incomplete raw dataset", async () => {
+    const newDataset = {
+      ...TestData.RawWrong_1,
+      ownerGroup: "group1",
+    };
+
+    return request(appUrl)
+      .post("/api/v3/Datasets")
+      .send(newDataset)
+      .set("Accept", "application/json")
+      .set({ Authorization: `Bearer ${accessTokenUser1}` })
+      .expect("Content-Type", /json/)
+      .then((res) => {
+        res.statusCode.should.not.be.equal(200);
+        res.body.should.not.have.property("pid");
+      });
+  });
+
+  it("0602: user with create dataset groups only can not add a new raw dataset with specified pid", async () => {
+    const datasetWithPid = {
+      ...TestData.RawCorrect,
+      pid: uuidv4(),
+      ownerGroup: "group1",
+    };
+
+    return request(appUrl)
+      .post("/api/v3/Datasets")
+      .send(datasetWithPid)
+      .set("Accept", "application/json")
+      .set({ Authorization: `Bearer ${accessTokenUser1}` })
+      .expect("Content-Type", /json/)
+      .then((res) => {
+        res.statusCode.should.not.be.equal(200);
+        res.body.should.not.have.property("pid");
+      });
+  });
+
+  it("0603: user with create dataset groups only cannot add a new raw dataset with specified invalid pid", async () => {
+    const datasetWithPid = {
+      ...TestData.RawCorrect,
+      pid: "this-is-invalid-pid-1",
+      ownerGroup: "group1",
+    };
+
+    return request(appUrl)
+      .post("/api/v3/Datasets")
+      .send(datasetWithPid)
+      .set("Accept", "application/json")
+      .set({ Authorization: `Bearer ${accessTokenUser1}` })
+      .expect("Content-Type", /json/)
+      .then((res) => {
+        res.statusCode.should.not.be.equal(200);
```

Comment:
These test cases should be part of authentication process, so  I'd suggest to use equal to 403 status for all the similar tests, to make the test more precise.